pixeltable/func/query_template_function.py

```diff
@@ -112,3 +123,85 @@ def make_query_template(py_fn: Callable, param_types: Optional[list[ts.ColumnTyp
     else:
         assert len(args) == 0 and len(kwargs) == 1 and 'param_types' in kwargs
         return lambda py_fn: make_query_template(py_fn, kwargs['param_types'])
+
+
+def retrieval_tool(
+    table: catalog.Table,
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+    parameters: Optional[Iterable[Union[str, exprs.ColumnRef]]] = None,
+    limit: Optional[int] = 10,
+) -> func.QueryTemplateFunction:
+    """
+    Constructs a retrieval tool for the given table. The retrieval tool is a function `f` whose parameters are
+    columns of the table and whose return value is a list of rows from the table. The return value of
+    ```python
+    f(col1=x, col2=y, ...)
+    ```
+    will be a list of all rows from the table that match the specified arguments.
+
+    Args:
+        table: The table to use as the dataset for the retrieval tool.
+        name: The name of the tool. If not specified, then the name of the table will be used by default.
+        description: The description of the tool. If not specified, then a default description will be generated.
+        parameters: The columns of the table to use as parameters. If not specified, all data columns
+            (non-computed columns) will be used as parameters.
```

Comment:
Why exclude computed columns? I can how those would be useful (eg, "transcript contains the word xyz").

Reply:
Or even just "object_class == 'car'", if object_class is a computed column for 'detections[0].class]'.